compiler/src/iree/compiler/DispatchCreation/Passes.h

```diff
@@ -17,6 +17,8 @@
 
 namespace mlir::iree_compiler::DispatchCreation {
 
+enum class EncodingOptions { MatmulK, Default };
```

Comment:
As discussed offline, I think `Generic` is better than `Default` because it encodes as much information as possible. Please also add a doc for the two members. E.g., here is my understanding of the strategy:

- Generic: use EncodingAttr which encodes as much information as possible.
- MatmulK: only encodes reduction dimensions in the encoding.